README.md

```diff
@@ -1,75 +1,163 @@
-## [x] Generating information about consumer in pacts
+```bash
+yarn add --dev pact-gen-ts
+```
 
-In **pacts.config.js**:
+Next you should create a minimal `pacts.config.js` configuration file in the root directory:
 
 ```js
 module.exports = {
-    consumer: 'some-consumer',
+    consumer: 'consumer-name',
+    providers: [
+        {
+            provider: 'some-provider',
+            files: ['src/api/**/*.ts'],
+        },
+    ],
 };
 ```
 
-## [x] Generating information about provider in pacts
+where `files` property will be an array of glob patterns pointing to API functions definitions.
+
+After that pact-gen-ts is ready, now you need to mark all API functions which will be analysed:
+
+```ts
+/**
+ * @pact
+ */
+function fetchComments() {
+    // ...
+}
+```
+
+Last thing is to execute command:
+
+```bash
+pact-gen-ts
+```
+
+which does the analysis and generates pacts in JSON format inside (by default) `./pacts` directory.
 
-In **pacts.config.js**:
+## Compatibility with TypeScript
+
+Due to TypeScript's occasional changes to its compiler API and not following semantic versioning in their releases, latest versions of pact-gen-ts can only guarantee compatibility with latest versions of TypeScript.
+
+If you're limited to historical versions of TypeScript, you should install a corresponding version of pact-gen-ts. Below table presents what TS versions pact-gen-ts will work with:
+
+| pact-gen-ts | TypeScript |
+| ----------- | ---------- |
+| 0.8         | 4.1 - 4.2  |
+| 0.9 - x     | \>= 4.5    |
+
+## Configuration
+
+Pact-gen-ts uses configuration stored in `pacts.config.js` file in project's root directory:
 
 ```js
 module.exports = {
+    consumer: 'consumer-name',
+    buildDir: 'pacts',
+    verbose: true,
     providers: [
         {
-            provider: 'some-provider',
-            files: ['src/api/**/*.ts'],
+            provider: 'provider-name',
+            files: ['src/api/firstProvider/*.ts'],
+            queryArrayFormat: 'indices',
+            requestHeaders: {
+                authorization: 'auth',
+            },
+            responseHeaders: {
+                'Content-Type': 'application/json',
+            },
         },
     ],
 };
 ```
 
-### [x] Specify multiple providers
+### Options
 
-## [x] Add metadata and pact specification into pacts
+| Option      | Description                                                                            |
+| ----------- | -------------------------------------------------------------------------------------- |
+| `consumer`  | Consumer's name                                                                        |
+| `buildDir`  | Directory where generated pacts will be placed                                         |
+| `verbose`   | If set to `true` additional information during pacts generating process will be logged |
+| `providers` | Array of options for each provider                                                     |
 
-Pact Specification in version 2.0.0
```

Comment:
Quick idea - document `providers` in the same table as other options using dot-notation for increased readability.


| Option      |  Required  |   Default   | Description                                                                            |
|-------------|:----------:|:-----------:|----------------------------------------------------------------------------------------|
| `consumer`  |    Yes     |      -      | Consumer's name                                                                        |
| `providers` |    Yes     |      -      | Array of options for each provider                                                     |
| `providers[].provider` |    Yes     |      -      |  Provider's name                                                     |
| `providers[].files` |    Yes     |      -      |    Array of glob patterns where API functions are defined                                                   |
| `providers[]...` |    Yes     |      -      |                                                       |
| `buildDir`  |     No     |  `./pacts`  | Directory where generated pacts will be placed                                         |
| `verbose`   |     No     |   `false`   | If set to `true` additional information during pacts generating process will be logged